src/components/FileList.tsx

```diff
@@ -0,0 +1,574 @@
+    event.persist()
+    event.preventDefault()
+    this.setState(
+      {
+        contextMenuTypeX: typeX,
+        contextMenuTypeY: typeY,
+        contextMenuFile: file
+      }, 
+      () => this.state.contextMenuUnstaged.open(event.clientX, event.clientY)
+    )
+  }
+
+  /** Handle right-click on an untracked file */
+  contextMenuUntracked = (event, typeX: string, typeY: string, file: string) => {
+    event.persist()
+    event.preventDefault()
+    this.setState(
+      {
+        contextMenuTypeX: typeX,
+        contextMenuTypeY: typeY,
+        contextMenuFile: file
+      }, 
+      () => this.state.contextMenuUntracked.open(event.clientX, event.clientY)
+    )
+  }
+  
+  /** Handle input inside commit message box */
+  handleChange = (event: any) : void => {
+    if (event.target.value && event.target.value !== '') {
+      this.setState(
+        {
+          commitMessage: event.target.value,
+          disableCommit: false
+        }
+      )
+    } else {
+      this.setState(
+        {
+          commitMessage: event.target.value,
+          disableCommit: true
+        }
+      )
+    }
+  }
+
+  /** Initalize commit message input box */
+  initializeInput = () : void => {
+    this.setState(
+      {
+        commitMessage: '',
+        disableCommit: true
+      }
+    )
+  }
+
+  /** Prevent enter key triggered 'submit' action during commit message input */
+  onKeyPress(event) : void {
+    if (event.which === 13) {
+      event.preventDefault()
+      this.setState({commitMessage: this.state.commitMessage + '\n'})
+    }
+  }
+
+  /** Toggle display of staged files */
+  displayStaged = () : void => {
+    this.setState({showStaged: !this.state.showStaged})
+  }
+
+  /** Toggle display of unstaged files */
+  displayUnstaged = () : void => {
+    this.setState({showUnstaged: !this.state.showUnstaged})
+  }
+
+  /** Toggle display of untracked files */
+  displayUntracked = () : void => {
+    this.setState({showUntracked: !this.state.showUntracked})
+  }
+
+  /** Update state of commit message input box */
+  updateCommitBoxState(disable: boolean, numberOfFiles: number) {
+    if (disable) {
+      if (numberOfFiles === 0) {
+        return classes(stagedCommitButtonStyle, stagedCommitButtonDisabledStyle)
+      } else {
+        return classes(stagedCommitButtonStyle, stagedCommitButtonReadyStyle)
+      } 
+    } else {
+      return stagedCommitButtonStyle
+    }
+  }
+
+/** Open a file in the git listing */
+async openListedFile(typeX: string, typeY: string, path: string, app: JupyterLab) {
+  if (typeX === 'D' || typeY === 'D') {
+    showDialog(
+      {
+        title: 'Open File Failed',
+        body: "This file has been deleted!",
+        buttons: [Dialog.warnButton({ label: 'OK' })]
+      }
+    ).then(result => {
+      if (result.button.accept) {
+        return
+      }
+    })
+   return
+  } try {
+    let leftSidebarItems = app.shell.widgets('left')
```

Comment:
Another personal preference: use `const` unless you *need* to reassign a variable